### ISSUE TITLE
Footnotes: use of code to group table and table notes

### ISSUE THREAD

**Issue by RubyDo** [wet-boew/wet-boew#6212]
Hello, 
It would be great if we see an example of coding table notes on the WET 4 examples page. 
Some manuals include table notes. Table notes follow the table and are not part of the end of the page. 

At CFIA we use the same footnote project to code table notes and use the &lt;figure&gt; element to represent the table the footnote as a single unit from the main flow. 
e.g.  
&lt;figure&gt;<br>
&lt;table&gt;<br>
&lt;!-- table content--&gt;<br>
&lt;/table&gt;<br>
&lt;div class=&quot;clearfix&quot;&gt; &lt;/div&gt;<br>
&lt;aside class=&quot;wb-fnote&quot; role=&quot;note&quot;&gt;<br>
&lt;h2 id=&quot;fn-table1&quot;&gt;Table Notes&lt;/h2&gt;<br>
&lt;dl&gt;<br>
&lt;!-- place &lt;dt&gt; and &lt;dd&gt; here --&gt;<br>
&lt;/dl&gt;<br>
&lt;/aside&gt;<br>
&lt;/figure&gt;

**Comment by RubyDo** [comment]
I reworked the example to use only role="note" and have the table followed by the table notes. 
Note Numbers used only once within the same page. 
<pre> 
&lt;table&gt; .... &lt;/table&gt;<br>
&lt;div role=&quot;note&quot;&gt;<br>
&lt;p&gt;&lt;strong&gt;Table Notes&lt;/strong&gt;&lt;/p&gt;<br>
&lt;dl&gt;<br>
&lt;dt&gt;Table  note 1&lt;/dt&gt;<br>
&lt;dd id=&quot;fn1&quot;&gt;<br>
&lt;p&gt;Cars are vehicle that human use.&lt;/p&gt;<br>
&lt;p&gt;&lt;a href=&quot;#fn1-rf&quot;&gt;Return to table note&lt;/span&gt;&amp;nbsp;1 referrer&lt;/span&gt;&lt;/a&gt; &lt;/p&gt;<br>
&lt;/dd&gt;<br>
....<br>
&lt;/dl&gt;
</pre>

@thomasgohard @EricDunsworth  what do you think?

**Comment by dilborg** [comment]
I'm unclear as to why the Notes are being excluded from the TABLE container and added below the Table in a separate DIV.

The purpose of the TFOOT is really for Footnotes, and Endnotes for the Table. 

If you don't use TFOOT, you would lose any responsive elements for the Notes which would be inherited from your Table.  So . . . no ability to stripe your Notes . . . no flow around Notes for exceptionally long tables . . . borders . . . etc . . . etc . . . 

In an ideal and perfect world, authors would make the Tables self explanatory =)

**Comment by RubyDo** [comment]
@dilborg I don't see <tfoot> for table notes. 
- <tfoot> defines a set of rows summarizing the columns of a table. The notes are not summaries for the columns, they are additional information about specific cells. 
- In my proposal, the association of cell and note list item is done through the use of bookmark. 
The  role="note" adds a bit of semantic to define the purpose of the notes decision 
Rabab

**Comment by dilborg** [comment]
We are talking about a data table and not a layout table am I correct?

And the question being asked was 'accessible footnotes'?  

I discourage footnotes in a table and have sent back many a table for lack of clarity when they are used.  However if they are there and if I want to make the accessible under WxT4 I use TFOOT.

English: 
              <tfoot>
                <tr>
                  <td colspan="3">
	         <aside class="wb-fnote"  role="note">
                       <h6 class="wb-inv" id="fnb-tb1">Footnotes</h6>
                       <dl>
                         <dt>Footnote 1</dt>
                         <dd id="fnb1-tb1">
                           <p>For text</p>
                           <p class="fn-rtn"><a href="#fnb1-tb1-ref"><span class="wb-inv">Return to footnote&#160;</span>1<span class="wb-inv"> referrer</span></a>
</p>
                        </dd>
                      </dl>
                    </aside>
                  </td>
                </tr>
              </tfoot>

French:
<tfoot>
	<tr>
		<td colspan="3">
		<aside class="wb-fnote">
		<h6 class="wb-inv" id="fnb-tb1">Note de bas de page</h6>
	<dl>
		<dt>Note de bas de page 1</dt>
		<dd id="fnb1">
		<p>NOTETEXT</p>
	<p class="fn-rtn">
		<a href="#fnb1-ref">
		<span class="wb-inv">Retour à la référence de la note de bas de page </span>
		1</a></p>
	</dd>
	</dl>
	</aside>
	</td>
	</tr>
</tfoot>

**Comment by dilborg** [comment]
We are talking about a data table and not a layout table am I correct?

And the question being asked was 'accessible footnotes'?  

I discourage footnotes in a table and have sent back many a table for lack of clarity when they are used.  However if they are there and if I want to make the accessible under WxT4 I use TFOOT.

English: 
```html
              <tfoot>
                <tr>
                  <td colspan="3">
	         <aside class="wb-fnote"  role="note">
                       <h6 class="wb-inv" id="fnb-tb1">Footnotes</h6>
                       <dl>
                         <dt>Footnote 1</dt>
                         <dd id="fnb1-tb1">
                           <p>For text</p>
                           <p class="fn-rtn"><a href="#fnb1-tb1-ref"><span class="wb-inv">Return to footnote&#160;</span>1<span class="wb-inv"> referrer</span></a>
</p>
                        </dd>
                      </dl>
                    </aside>
                  </td>
                </tr>
              </tfoot>


French:
```html
<tfoot>
	<tr>
		<td colspan="3">
		<aside class="wb-fnote">
		<h6 class="wb-inv" id="fnb-tb1">Note de bas de page</h6>
	<dl>
		<dt>Note de bas de page 1</dt>
		<dd id="fnb1">
		<p>NOTETEXT</p>
	<p class="fn-rtn">
		<a href="#fnb1-ref">
		<span class="wb-inv">Retour à la référence de la note de bas de page </span>
		1</a></p>
	</dd>
	</dl>
	</aside>
	</td>
	</tr>
</tfoot>

**Comment by hsrudnicki** [comment]
@RubyDo @pjackson28 @LaurentGoderre @thomasgohard and others: Are you ready to close this issue?

**Comment by RubyDo** [comment]
I am still curious about @LaurentGoderre  and @thomasgohard  opinion about the best way of coding table notes. It is something existing in many gov. pages and it would be great to be consistent.

As for CFIA, we group the notes and table using figure element based on W3C <a href="http://www.w3.org/TR/html5/common-idioms.html#footnotes">4.12.5 Footnotes</a>   
"A figure element is used to give a single legend to the combination of the table and its footnotes." and benefit from the footnote plugin. 
Here is the <a href="http://www.inspection.gc.ca/about-the-cfia/help/site-map/wet-4-0-html-technical-requirements/eng/1366899144603/1366899315072?pedisable=false#tablenotes-section">Table note code</a> we use.

**Comment by hsrudnicki** [comment]
Made "Going Stale" and closed due to no further discussion/status updates, and if otherwise please again update this issue's status, thank you

**Comment by samer73** [comment]
Hello everyone,

I would like to ask for the best way to create an accessible Tablenote? Any updates to the above discussion?

Thank you!
